drivers/display/display_st7796s.c

```diff
@@ -118,7 +118,13 @@ static int st7796s_get_pixelfmt(const struct device *dev)
 	 * and vice versa.
 	 */
 	if (config->dbi_config.mode == MIPI_DBI_MODE_8080_BUS_8_BIT) {
-		if (config->madctl & ST7796S_MADCTL_BGR) {
+		/*
+		 * Similar to the handling for other interface modes,
+		 * invert the reported pixel format if rgb_is_inverted"
+		 * is enabled
+		 */
+		if (!(config->madctl & ST7796S_MADCTL_BGR) !=
+		    !config->rgb_is_inverted) {
```

Comment:
```suggestion
                if ((bool)(config->madctl & ST7796S_MADCTL_BGR) != config->rgb_is_inverted) {
```

expression seems redundant with the double negative, also I'm not expert on the C standard but I think the first operand might need to be a boolean, at least better safe than sorry

Reply:
Casting certainly won't hurt. I *think* the reason the logic was written like this initially was that the `!` is going to effectively cast the value to a boolean (at least per my understanding). I'll update to make this change, and fix the logic on the lines below to match it

Reply:
i see, makes sense, still think it's not as clear of a way to write it though which is more important than saving a couple characters

Reply:
Sure- I changed the logic to match your suggestion.